Library/Homebrew/tap.rb

```diff
@@ -118,6 +119,12 @@ def git?
     path.git?
   end
 
+  # The current git branch of this {Tap}.
+  def git_branch
+    raise TapUnavailableError, name unless installed?
```

Comment:
How about punting the `TapUnavailableError` somehow into the `GitClient` so you can avoid repeating this. All it's checking is `path.directory?` which you could do in the client.